### ISSUE TITLE
Semantic versioning

### ISSUE THREAD

**Issue by Doomsdayrs** [ATLauncher/ATLauncher#555]
**Is your feature request related to a problem? Please describe.**
The project uses a strange versioning scheme currently

**Describe the solution you'd like**
Use [semantic versioning](https://semver.org/)

**Additional context**
The reason this comes up is to make it easier to understand what each number means with the versioning

**Comment by RyanTheAllmighty** [comment]
Yup it's kind of a bastardised version of semantic. Ignoring the first number in the version it is essentially semantic.

I've wanted to move to strictly semantic with strictly automated changelogs and versioning simply to make my life easier, the only reason I hadn't was due to that first number in the release, always wanting that Major version to only change for complete rewrites.

But now that ATLauncher Next rewrite is 99% not going to happen, I think I'll probbaly look at doing this. And thinking about it anyway, there's no reason the major version number would change anyway unless something backwards incompatable changed anyway.

**Comment by Doomsdayrs** [comment]
Woah, I did not know about that.

I only posted this because it would make it easier to make the flatpak version (less work for me to figure out regex)

**Comment by Doomsdayrs** [comment]
Since its the perfect time, How about consider either v3.4.14 or v3.5.0 for the next release?

**Comment by RyanTheAllmighty** [comment]
There's a few things I need to do before this gets resolved:

- [ ] Check the backend can handle 3 dot versions
- [ ] Check the launcher can handle 3 dot versions when checking for updates
- [ ] Find a library to enforce conventional commits through hooks installed through gradle preferably
- [ ] Enforce conventional commits at the ci level
- [ ] Find a library/script that can auto generate changelogs from conventional commits (likely JS based as there is a large ecosystem there for it) to run on CI master builds
